Marlin/src/feature/fwretract.cpp

```diff
@@ -106,6 +101,8 @@ void FWRetract::retract(const bool retracting
     SERIAL_ECHOLNPAIR("hop_amount ", hop_amount);
   //*/
 
+  stepper.synchronize();  // Wait for buffered moves to complete
```

Comment:
This is the only change needed. The classes that only deserve one instance are all-static singletons for consistency. Meanwhile the creation of a "dummy" instance is purely to obtain the somewhat cleaner-looking dot-notation. The linkage is done identically to `Stepper::`. The advantage of all-static methods is that there is no hidden `this` parameter passed in the method call — thus the code is slightly smaller and faster, just as if these methods were extern/static functions.